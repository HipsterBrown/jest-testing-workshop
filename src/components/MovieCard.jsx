import React from 'react';
import PropTypes from 'prop-types';
import {Link} from 'redux-little-router';

const MovieCard = ({link, poster, title}) => (
  <div className="tc">
    <img src={poster} alt={title} className="db w-100 ba b--black-10 mb3" />
    {Boolean(link) ? (
      <Link href={link} className="link underline-hover near-black f5 fw6">
        {title}
      </Link>
    ) : (
      <p className="f5 fw6 near-black">{title}</p>
    )}
  </div>
);

MovieCard.propTypes = {
  link: PropTypes.string,
  poster: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
};

export default MovieCard;
