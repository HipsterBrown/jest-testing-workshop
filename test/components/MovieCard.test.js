import React from 'react';
import renderer from 'react-test-renderer';
import MockRouterProvider from 'helpers/MockRouterProvider';
import MovieCard from 'components/MovieCard';

describe('<MovieCard />', () => {
  test('displays card without link prop', () => {
    const props = {
      poster: '/test.png',
      title: 'Test',
    };

    expect(
      renderer.create(<MovieCard {...props} />).toJSON(),
    ).toMatchSnapshot();
  });

  test('displays card with link prop', () => {
    const props = {
      link: 'test-link.com',
      poster: '/test.png',
      title: 'Test',
    };

    expect(
      renderer
        .create(
          <MockRouterProvider>
            <MovieCard {...props} />
          </MockRouterProvider>,
        )
        .toJSON(),
    ).toMatchSnapshot();
  });
});
