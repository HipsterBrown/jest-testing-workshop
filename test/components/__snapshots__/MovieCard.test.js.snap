// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieCard /> displays card with link prop 1`] = `
<div
  className="tc"
>
  <img
    alt="Test"
    className="db w-100 ba b--black-10 mb3"
    src="/test.png"
  />
  <a
    className="link underline-hover near-black f5 fw6"
    href="test-link.com"
    onClick={[Function]}
    target={undefined}
  >
    Test
  </a>
</div>
`;

exports[`<MovieCard /> displays card without link prop 1`] = `
<div
  className="tc"
>
  <img
    alt="Test"
    className="db w-100 ba b--black-10 mb3"
    src="/test.png"
  />
  <p
    className="f5 fw6 near-black"
  >
    Test
  </p>
</div>
`;
